questionary/prompts/date.py

```diff
@@ -0,0 +1,674 @@
+            value can not be submitted until the validator accepts it (e.g. to check
+        minimum password length). This can either be a function accepting the input and
+        returning a boolean, or an class reference to a subclass of the prompt toolkit
+        Validator class. Defaults to None.
+        completer (Completer, optional): A :class: `prompt_toolkit.completion.Completer`
+            yielding completions for user input. If None, :class: `FullDateCompleter` is
+            used. Defaults to None. Defaults to None.
+        style (Style, optional): A custom color and style for the question parts. You
+            can configure colors as well as font types for different elements.
+            Defaults to None.
+        date_format (str, optional): Format determining the format that is to be used
+            for parsing :class: `datetime.date`. If set to None (and no ``completer`` was
+            set) completions of :class: `SimpleDateCompleter` are deactivated.Defaults to
+            ``ISO8601``.. Defaults to ISO8601.
+        print_date_format (bool): If set to True, ``date_format`` is printed on the
+            right of the prompt. Defaults to True.
+        parser (Callable[[str], AnyDate], optional): A callable that parses
+                a string into a :class: `datetime.date` or :class: `datetime.date`, e.g.
+                the ones from `dateparser`_ or `dateutil`_. If None, set to
+                :function: `custom_date_parser`. Defaults to None.
+                Has no effect is ``no_extra_parser`` is set to True.
+        no_extra_parser (bool): If True, completion and validation using the ``parser`` is
+            not performed. Defaults to False.
+        return_date_object (bool): If True, a parsed date object is returned. Else, string
+            the text input is returned. Defaults to True.
+        complete_style (CompleteStyle): How autocomplete menu would be shown, it could
+            be ``COLUMN`` ``MULTI_COLUMN`` or ``READLINE_LIKE`` from
+            :class:`prompt_toolkit.shortcuts.CompleteStyle`. Defaults to
+            CompleteStyle.MULTI_COLUMN.
+
+    Returns:
+        :class:`Question`: Question instance, ready to be prompted (using ``.ask()``).
+
+    Example:
+        >>> import questionary
+        >>> questionary.date("Type a date: ").ask()
+        ? Type a date: 2021-01-01
+        '2021-01-01'
+
+    .. _dateparser: https://github.com/scrapinghub/dateparser
+    .. _dateutil: https://github.com/dateutil/dateutil
+    """
+    # delimeter used to separate year, month and day
+    if isinstance(date_format, str):
+        delimeter: str = date_format[-3]
+    else:
+        delimeter = " "
+
+    parser = parser or custom_date_parser
+    if no_extra_parser:
+        parser = None
+
+    merged_style = merge_styles([DEFAULT_STYLE, style])
+
+    if print_date_format:
+        rprompt = to_formatted_text(
+            f"Date format: {date_format}", style="bg: ansigreen bold"
```

Comment:
Can these styles use a class (or a preexisting class) so that they can be customised.

Reply:
```suggestion
if print_date_format:
        rprompt = to_formatted_text(
            f"Date format: {date_format}", style="class:information"
        )
```